build.sbt

```diff
@@ -187,6 +187,7 @@ val `polynote-kernel` = project.settings(
     "com.github.javaparser" % "javaparser-symbol-solver-core" % versions.javaparser,
     "org.scalamock" %% "scalamock" % "4.4.0" % "test"
   ),
+  dependencyOverrides += "com.chuusai" %% "shapeless" % "2.3.2",
```

Comment:
Do we need to pick a specific version based on what matches Spark?

Reply:
yeah, I think so. I wish there were a way to tell sbt, "use the version that you get from Spark, not a later version" but there isn't as far as I know.

Reply:
Is there no like "provided" dependency? compileOnly or whatever?

Reply:
There is `provided`... but I don't see how that would solve this issue?

Reply:
Oh, by "compiletime/runtime" I mean _cell_ compile time and _cell_ runtime. The issue is that the shapeless in `polynote-kernel-assembly` (which needs to be there in the absence of Spark) gets used during compile time for some reason, and its macros are what get executed. But during runtime, Spark's shapeless gets used, which doesn't have the code that the macro referred to.